op-challenger/fault/cmd/examples/full.go

```diff
@@ -0,0 +1,38 @@
+package examples
+
+import (
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/fault"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+func FullGame() {
+	log.Root().SetHandler(
+		log.LvlFilterHandler(log.LvlInfo, log.StreamHandler(os.Stdout, log.TerminalFormat(true))),
+	)
+
+	canonical := "abcdefgh"
+	disputed := "abcdexyz"
+	maxDepth := 3
```

Comment:
nit:
```suggestion
	maxDepth uint64 := 3
```
and then you don't need to cast it multiple times below.

Reply:
or use a `const` for this and the strings above, then you never have to convert `maxDepth`.

Reply:
Switched to starting it as uint64. We definitely use both uint64 & int's which makes it kinda annoying. I'll have to go through & unify the types.